src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -0,0 +1,865 @@
+  /*
+  {
+    kind: String // the type of selection, one of PICKER_KINDS (PRESET, RELATIVE, ABSOLUTE)
+    preset: {
+      label: String // the label of the selected preset
+      offset: Number // the offset in minute
+    },
+    relative: {
+      start: Date // the start point in time
+      end: Date // the end point in time
+      lastNumber: Number // quantity of interval
+      lastInterval: String // one of INTERVAL_VALUES
+      relativeToWhen: String // one of RELATIVE_VALUES, indicates to what point in time the selection is relative to
+      relativeToTime: String // in the HH:MM format
+    },
+    absolute: {
+      start: Date // the start point in time
+      end: Date // the end point in time
+      startDate: String // start date in the mask or default format
+      startTime: String // in the HH:MM format
+      endDate:  // end date in the mask or default format
+      endTime: String // in the HH:MM format
+    },
+  } */
+  onApply: PropTypes.func,
+  /** All the labels that need translation */
+  i18n: PropTypes.shape({
+    toLabel: PropTypes.string,
+    toNowLabel: PropTypes.string,
+    calendarLabel: PropTypes.string,
+    presetLabels: PropTypes.string,
+    intervalLabels: PropTypes.string,
+    relativeLabels: PropTypes.string,
+    customRangeLinkLabel: PropTypes.string,
+    customRangeLabel: PropTypes.string,
+    relativeLabel: PropTypes.string,
+    lastLabel: PropTypes.string,
+    invalidNumberLabel: PropTypes.string,
+    relativeToLabel: PropTypes.string,
+    absoluteLabel: PropTypes.string,
+    startTimeLabel: PropTypes.string,
+    endTimeLabel: PropTypes.string,
+    applyBtnLabel: PropTypes.string,
+    cancelBtnLabel: PropTypes.string,
+    backBtnLabel: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  defaultValue: null,
+  dateTimeMask: 'YYYY-MM-DD HH:mm',
+  presets: PRESET_VALUES,
+  intervals: [
+    {
+      label: 'minutes',
+      value: INTERVAL_VALUES.MINUTES,
+    },
+    {
+      label: 'hours',
+      value: INTERVAL_VALUES.HOURS,
+    },
+    {
+      label: 'days',
+      value: INTERVAL_VALUES.DAYS,
+    },
+    {
+      label: 'weeks',
+      value: INTERVAL_VALUES.WEEKS,
+    },
+    {
+      label: 'months',
+      value: INTERVAL_VALUES.MONTHS,
+    },
+    {
+      label: 'years',
+      value: INTERVAL_VALUES.YEARS,
+    },
+  ],
+  relatives: [
+    {
+      label: 'Yesterday',
+      value: RELATIVE_VALUES.YESTERDAY,
+    },
+    {
+      label: 'Today',
+      value: RELATIVE_VALUES.TODAY,
+    },
+    {
+      label: '',
+      value: '',
+    },
+  ],
+  expanded: false,
+  disabled: false,
+  showRelativeOption: true,
+  onCancel: null,
+  onApply: null,
+  i18n: {
+    toLabel: 'to',
+    toNowLabel: 'to Now',
+    calendarLabel: 'Calendar',
+    presetLabels: 'Last 30 minutes|Last 1 hour|Last 6 hours|Last 12 hours|Last 24 hours',
+    intervalLabels: 'minutes|hours|days|weeks|months|years',
+    relativeLabels: 'Yesterday|Today',
```

Comment:
An array of strings would be more appropriate for these. I don't think it would conflict with i18n libraries in applications? That or they could be a function that returns an array of strings.